File: src/test/resources/ambiguity/FieldAndMethodWithSameNameGeneric.cs
namespace ambiguity {
    public class FieldAndMethodWithSameNameGeneric<T> {
        public virtual int getNow() {
            return now__;
        }

        public virtual void setNow(int now) {
            this.now__ = now;
        }

        public int now__;

        public virtual int now() {
            return now__;
        }

        public virtual void work() {
            System.Console.Out.WriteLine("now: " + now__);
            System.Console.Out.WriteLine("now: " + this.now__);
            System.Console.Out.WriteLine("now(): " + now());
            now__ = 100;
            this.now__ = 100;
        }
    }
}

File: src/test/resources/ambiguity/FieldAndMethodWithSameNameGeneric.java
package ambiguity;

public class FieldAndMethodWithSameNameGeneric<T> {
    public int getNow() {
        return now;
    }

    public void setNow(int now) {
        this.now = now;
    }

    public int now;

    public int now() {
        return now;
    }

    public void work() {
        System.out.println("now: " + now);
        System.out.println("now: " + this.now);
        System.out.println("now(): " + now());
        now = 100;
        this.now = 100;
    }

}
